p2p/net/conn/transport.go

```diff
@@ -0,0 +1,191 @@
+package conn
+
+import (
+	"net"
+
+	ma "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr"
+	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
+	mautp "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net/utp"
+	reuseport "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-reuseport"
+)
+
+type Transport interface {
+	manet.Listener
+	ProtoDialer
+}
+
+type ProtoDialer interface {
+	Dial(raddr ma.Multiaddr) (manet.Conn, error)
```

Comment:
My thought was that a transport should be bound to a local address. I think it makes it easier to reason about.